### DISCUSSION TITLE
Inputs.required not enforced with no default?

### DISCUSSION THREAD

**Discussion by gjadi-mej** [community/community#26777] [Actions]
<p>Hi, I’m following the <em>GitHub Actions: Writing JavaScript Actions</em> course and I have a question regarding the <code>inputs.required</code> parameter.</p>
<p>I updated my <code>action.yml</code> file so that an input is required but no default is provided:</p>
<pre><code class="lang-yaml">inputs:
  first-greeting:
    description: "who would you like to greet in the console"
    required: true
</code></pre>
<p>In the workflow, I don’t provide the input either:</p>
<pre><code class="lang-auto">      - name: hello-action
        uses: ./.github/actions/hello-world
</code></pre>
<p>Observed results: the action is successfully ran:</p>
<pre><code class="lang-auto">Run ./.github/actions/hello-world
Hello 
Hello Mona the Octocat
</code></pre>
<p>Expected results: an error is raised.</p>
<p>Here is the repo+commit corresponding: <a href="https://github.com/gjadi-mej/writing-javascript-actions/tree/0dcc416d922c1572124153b9b949fec172b59aaf/.github/workflows" class="inline-onebox" rel="noopener nofollow ugc">writing-javascript-actions/.github/workflows at 0dcc416d922c1572124153b9b949fec172b59aaf · gjadi-mej/writing-javascript-actions · GitHub</a><br>
Here is the action log (sorry only 2 links allowed for a first post): gjadi-mej/writing-javascript-actions/actions/runs/1345241384</p>
<p>Here is the documentation <a href="https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#inputsinput_idrequired" rel="noopener nofollow ugc">(source)</a>:</p>
<blockquote>
<p><strong>Required</strong> A <code>boolean</code> to indicate whether the action requires the input parameter. Set to <code>true</code> when the parameter is required.</p>
</blockquote>
<p>The doc doesn’t say anything about raising an error but, to me, that’s the meaning of the world <code>required</code>.<br>
I haven’t yet dig further into the javascript lib, and I imagine it’s possible to detect the empty input toraise an error from the <code>main.js</code> but that seems redundant with the <code>required</code> keyword.</p>
<p>Did I miss something?<br>
Thanks for the answers.</p>
<p>Best,</p>

**Comment by rcowsill** [comment]
<p>I had a quick look through <a href="https://github.com/actions/runner" rel="noopener nofollow ugc">actions/runner</a> and can’t see anything that implements the “required” functionality for action inputs.</p>
<p>For inputs that aren’t specified in the workflow the default is used if set, otherwise an empty string is used. There’s no check that a value was specified in the workflow for “required” inputs</p>

**Comment by gjadi-mej** [comment]
<p>Ok, thanks for checking this out.<br>
I will open an issue on the repository.</p>
<p>Best,</p>

**Comment by Barabas5532** [comment, marked as answer]
<p>This issue looks to be for the same problem you described: <a href="https://github.com/actions/runner/issues/1070" class="inline-onebox" rel="noopener nofollow ugc">Validate required inputs are set before invoking an action · Issue #1070 · actions/runner · GitHub</a></p>
